gammapy/spectrum/spectrum_dataset.py

```diff
@@ -0,0 +1,205 @@
+
+    @property
+    def parameters(self):
+        if self._parameters is None:
+            raise AttributeError("No model set for Dataset")
+        else:
+            return self._parameters
+
+    @property
+    def data_shape(self):
+        """Shape of the counts data"""
+        return self.counts.data.shape
+
+    def npred(self):
+        """Returns npred counts vector """
+        if self._predictor is None:
+            raise AttributeError("No model set for this Dataset")
+        model_npred = self._predictor.compute_npred().data.data
+        return model_npred
+
+    def likelihood_per_bin(self):
+        """Likelihood per bin given the current model parameters"""
+        on_stat_ = wstat(
+            n_on=self.ONcounts.data.data,
+            n_off=self.OFFcounts.data.data,
+            alpha=self.alpha,
+            mu_sig=self.npred(),
+        )
+        return np.nan_to_num(on_stat_)
+
+    def likelihood(self, parameters, mask=None):
+        """Total likelihood given the current model parameters.
+
+        Parameters
+        ----------
+        mask : `~numpy.ndarray`
+            Mask to be combined with the dataset mask.
+        """
+        if self.mask is None and mask is None:
+            stat = self.likelihood_per_bin()
+        elif self.mask is None:
+            stat = self.likelihood_per_bin()[mask]
+        elif mask is None:
+            stat = self.likelihood_per_bin()[self.mask]
+        else:
+            stat = self.likelihood_per_bin()[mask & self.mask]
+
+        return np.sum(stat, dtype=np.float64)
+
+    @classmethod
+    def read_from_ogip(cls, filename):
+        """Read from OGIP files.
+
+        BKG file, ARF, and RMF must be set in the PHA header and be present in
+        the same folder.
+
+        Parameters
+        ----------
+        filename : str
+            OGIP PHA file to read
+        """
+        filename = make_path(filename)
+        dirname = filename.parent
+        on_vector = PHACountsSpectrum.read(filename)
+        rmf, arf, bkg = on_vector.rmffile, on_vector.arffile, on_vector.bkgfile
+
+        try:
+            energy_dispersion = EnergyDispersion.read(str(dirname / rmf))
+        except IOError:
+            # TODO : Add logger and echo warning
+            energy_dispersion = None
+
+        try:
+            off_vector = PHACountsSpectrum.read(str(dirname / bkg))
+        except IOError:
+            # TODO : Add logger and echo warning
+            off_vector = None
+
+        try:
+            effective_area = EffectiveAreaTable.read(str(dirname / arf))
+        except IOError:
+            # TODO : Add logger and echo warning
+            effective_area = None
+
+        return cls(
+            ONcounts=on_vector,
+            aeff=effective_area,
+            OFFcounts=off_vector,
+            edisp=energy_dispersion,
+            livetime=on_vector.livetime
+        )
+
+    def export_to_ogip(self, outdir=None, overwrite=False):
```

Comment:
Again I think it's fine to just call this `.write()`. Multiple I/O formats can be handled internally and must not be reflected in the API.

Reply:
Same comment and question as above.